### PR TITLE
[BugFix][Relay] Fix type relation for batch_matmul

### PR DESCRIPTION
**Pull request by lygztq**
I find the current implementation of `relay.nn.batch_matmul` will fail given input tensors with shape `(Any(), M, K)` and `(Any(), N, K)`.  This is caused by the wrong implementation of `batch_matmul`'s type relation. When the input tensor has a dynamic shape but not all dimensions are dynamic, this relation function will use `x`'s shape directly and output wrong results.

For example, given `x` with shape `(Any(), 16, 32)` and `y` with shape `(Any(), 16, 32)`, the output shape of current implementation is `(Any(), 16, 32)`.

### CODE REVIEW

**Comment by lygztq** [comment]
@mbrookhart @tqchen @zhiics @anijain2305 @ZihengJiang

File: tests/python/relay/test_op_level10.py [hunk 1]
```diff
@@ -380,10 +382,31 @@ def verify_dynamic_batch_matmul(x_shape, y_shape, out_shape, dtype="float32"):
 # TODO(mbrookhart): enable once VM supports heterogenous execution
 # @tvm.testing.uses_gpu
```
Comment (mbrookhart) [inline tests/python/relay/test_op_level10.py:383]: I thought I'd removed all of these TODO's. Oops! This should work on GPU now, I think

**Review by mbrookhart — APPROVED** [review]

File: tests/python/relay/test_op_level10.py (hunk 1 above)
Comment (lygztq) [reply tests/python/relay/test_op_level10.py:383]: I'm not sure, I think [this line](https://github.com/apache/tvm/blob/84359a97bdb896ddd2a74c65631a00da1ec5ec70/python/tvm/topi/nn/batch_matmul.py#L65-L65) (which is called by topi.cuda [here](https://github.com/apache/tvm/blob/90dce48de6ca74a439ed24fc3a3d80d57b305dd0/python/tvm/topi/cuda/batch_matmul.py#L32-L32)) will fail if `x.shape[2]` is static and `y.shape[2]` is dynamic.

By the way, the unit test in ci reported a strange error. I’m not sure if the error is related to this PR.

Comment (mbrookhart) [reply tests/python/relay/test_op_level10.py:383]: This is a flaky test that times out sometimes, rebase?

Comment (lygztq) [reply tests/python/relay/test_op_level10.py:383]: Everything is fine now, thanks.

**Pull request merged**